### DISCUSSION TITLE
Sort ENV variables by name

### DISCUSSION THREAD

**Discussion by heymartinadams** [vercel/vercel#4992] [Ideas]
Small feature idea: ability to sort ENV variables by name. Eventually, an app accumulates a ton of env variables and I’d like to sort them alphabetically to see if each of them are correct across all three environments (production, preview, development).
![image](https://user-images.githubusercontent.com/11673745/89468525-877efa00-d72c-11ea-9000-9dd20a320c9f.png)

![image](https://user-images.githubusercontent.com/11673745/89468462-5999b580-d72c-11ea-8615-b7192c6edb8d.png)

**Comment by styfle** [comment]
Thanks for the suggestion!

We have an internal Story 20828 tracking a redesign to the Environment Variables page so we'll be sure to add sorting too 👍

**Comment by verekia** [comment]
I posted a similar topic without seeing this one first. It has some formatted examples: #5118

**Comment by styfle** [comment]
Hi @heymartinadams @verekia @TaylorFacen

We now have a search feature where you can search for environment variables by name.

Do you still need to sort?

**Reply by heymartinadams** [reply]
Oh, thanks, @styfle! That’s awesome :)

No sort needed (on my end) — for unrelated reasons, we ended up going with using a secrets provider.

**Reply by verekia** [reply]
I still think an alphabetical sort makes more sense than a creation date sort. It would make it easier to find the variable in a long list. Searching is nice when you know exactly what you are looking for. It's not always the case. It also makes it easier to group together the same variables across environments. Right now we can get this kind of result:

ABC [Development] : abc-dev
SOMETHING_ELSE [Development] : something-else
ABC [Production, Preview] : abc-prod-preview

Would be nicer to have: 

ABC [Development] : abc-dev
ABC [Production, Preview] : abc-prod-preview
SOMETHING_ELSE [Development] : something-else
